do/loadbalancers.go

```diff
@@ -251,6 +277,30 @@ func (l *loadbalancers) buildLoadBalancerRequest(service *v1.Service, nodes []*v
 	}, nil
 }
 
+func (l *loadbalancers) waitActive(lbID string, timeout int, timespan int) (*godo.LoadBalancer, error) {
+	start := time.Now()
+	end := time.Second * time.Duration(timeout)
+	ts := time.Second * time.Duration(timespan)
+
+	for {
+		lb, _, err := l.client.LoadBalancers.Get(context.TODO(), lbID)
+		if err != nil {
+			return nil, err
+		}
+
+		if lb.Status == lbStatusActive {
+			return lb, nil
+		}
+		if lb.Status == lbStatusErrored {
+			return nil, fmt.Errorf("error creating balancer %q at Digital Ocean", lbID)
+		}
+		if end < time.Since(start) {
```

Comment:
Can we change this code to use ticker/contexts, I find its easier to understand. Something like:

```go
ctx, _ := context.WithTimeout(context.TODO(), time.Second*time.Duration(timeout))
for {
    select {
        case <-ctx.Done():
              return errors.New(...)
        case <- ticker.C:
              // check status here
    }
}
```

What do you think?